File: admin_service/src/AdminService.Api/Contracts/EventEnvelope.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace AdminService.Api.Contracts;

public sealed class EventEnvelope
{
    [JsonPropertyName("event_id")]
    public string EventId { get; set; } = string.Empty;

    [JsonPropertyName("event_type")]
    public string EventType { get; set; } = string.Empty;

    [JsonPropertyName("event_version")]
    public string EventVersion { get; set; } = "1.0";

    [JsonPropertyName("service")]
    public string Service { get; set; } = string.Empty;

    [JsonPropertyName("environment")]
    public string Environment { get; set; } = string.Empty;

    [JsonPropertyName("tenant")]
    public string Tenant { get; set; } = string.Empty;

    [JsonPropertyName("timestamp")]
    public DateTimeOffset Timestamp { get; set; } = DateTimeOffset.UtcNow;

    [JsonPropertyName("request_id")]
    public string? RequestId { get; set; }

    [JsonPropertyName("trace_id")]
    public string? TraceId { get; set; }

    [JsonPropertyName("correlation_id")]
    public string? CorrelationId { get; set; }

    [JsonPropertyName("user_id")]
    public string? UserId { get; set; }

    [JsonPropertyName("actor_id")]
    public string? ActorId { get; set; }

    [JsonPropertyName("aggregate_type")]
    public string AggregateType { get; set; } = string.Empty;

    [JsonPropertyName("aggregate_id")]
    public string AggregateId { get; set; } = string.Empty;

    [JsonPropertyName("payload")]
    public JsonElement Payload { get; set; }
}

File: calculator_service/src/main/java/com/microapp/calculator/kafka/EventEnvelope.java
package com.microapp.calculator.kafka;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.Instant;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record EventEnvelope(
        String eventId,
        String eventType,
        String eventVersion,
        String service,
        String environment,
        String tenant,
        Instant timestamp,
        String requestId,
        String traceId,
        String correlationId,
        String userId,
        String actorId,
        String aggregateType,
        String aggregateId,
        Map<String, Object> payload
) {
}
